src/navigation/index.js

```diff
@@ -0,0 +1,206 @@
+import React, { Component } from 'react';
+import _                    from 'underscore';
+import classNames           from 'classnames';
+
+// Components
+import BuiInputField from '../input-field';
+import BuiActionButton from '../action-button';
+
+const SearchItem = React.createClass({
+    propTypes: {
+        icon: React.PropTypes.string.isRequired,
+        name: React.PropTypes.string.isRequired,
+    },
+
+    render() {
+        return (
+            <div className='result-item'>
+                <div className='item-type'><i className={'type-icon fa ' + this.props.icon} /></div>
+                <div className='item-name'>{this.props.name}</div>
+            </div>
+        );
+    }
+});
+
+const Navigation = React.createClass({
+    propTypes: {
+        links: React.PropTypes.array.isRequired,
+        searchButton: React.PropTypes.bool,
+        logos: React.PropTypes.object,
+    },
+
+    getInitialState() {
+        return {
+            minimized: false,
+            searching: false,
+            searchValue: null,
+            hidden: false,
+        };
+    },
+
+    getDefaultProps() {
+        return {
+            searchButton: false,
+        }
+    },
+
+    _toggleMenuSize() {
+        this.setState({ minimized: !this.state.minimized });
+        this._closeSearch();
```

Comment:
You're setting state twice here, doesn't seem so optimal. I don't know if React handels this in a nice way?